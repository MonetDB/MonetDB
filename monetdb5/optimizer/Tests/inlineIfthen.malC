inline function user.foo(b:int):int;
   foo:= 0;
barrier go:= b<10;
   foo:= 1;
exit go;
   return foo;
end foo;

function qry();
	b:= bat.new(:int);
	bat.append(b,1);
	bat.append(b,10);
	bat.append(b,20);
	r:bat[:int]:= mal.multiplex("user","foo",b);
	io.print(r);
end qry;

optimizer.inline("user","qry");
optimizer.remap("user","qry");
mdb.List("user","qry");
