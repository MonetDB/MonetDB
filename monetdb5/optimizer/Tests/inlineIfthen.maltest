statement ok
inline function user.foo(b:int):int

statement ok
   foo:= 0

statement ok
barrier go:= b<10

statement ok
   foo:= 1

statement ok
exit go

statement ok
   return foo

statement ok
end foo

statement ok
function qry();
	b:= bat.new(:int);
	bat.append(b,1);
	bat.append(b,10);
	bat.append(b,20);
	r:bat[:int]:= mal.multiplex("user","foo",b);
	io.print(r);
end qry

statement ok
optimizer.inline("user","qry")

statement ok
optimizer.remap("user","qry")

