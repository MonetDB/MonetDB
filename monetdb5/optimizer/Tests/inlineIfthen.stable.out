stdout of test 'inlineIfthen` in directory 'monetdb5/optimizer` itself:


# 11:53:01 >  
# 11:53:01 >   mserver5 "--config=/ufs/mk/monet5/Linux/etc/monetdb5.conf" --debug=10 --set "monet_mod_path=/ufs/mk/monet5//Linux/lib/MonetDB5:/ufs/mk/monet5//Linux/lib/MonetDB5/lib:/ufs/mk/monet5//Linux/lib/MonetDB5/bin" --set "gdk_dbfarm=/ufs/mk/monet5//Linux/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/mk/monet5//Linux/var/MonetDB5/sql_logs"  --set mapi_open=true --set xrpc_open=true --set mapi_port=36453 --set xrpc_port=42260 --set monet_prompt= --trace --dbname=mTests_src_optimizer  inlineIfthen.mal
# 11:53:01 >  

# MonetDB 5 server v11.22.0
# This is an unreleased version
# Serving database 'mTests_monetdb5_optimizer', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 15.590 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2015 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.ins.cwi.nl:32947/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-19324/.s.monetdb.32947
# MonetDB/GIS module loaded
# Start processing logs sql/sql_logs version 52200
# Start reading the write-ahead log 'sql_logs/sql/log.4'
# Finished reading the write-ahead log 'sql_logs/sql/log.4'
# Finished processing logs sql/sql_logs
# MonetDB/SQL module loaded
# MonetDB/R   module loaded


# 21:52:28 >  
# 21:52:28 >  "mclient" "-lmal" "-ftest" "-Eutf-8" "--host=/var/tmp/mtest-23787" "--port=32128"
# 21:52:28 >  

function user.qry():void;               	#[0] (0)  0 
    b:bat[:int] := bat.new(:int);       	#[1] (0) CMDBATnew 1 <- 2 
    bat.append(b:bat[:int], 1:int);     	#[2] (0) BKCappend_val_wrap 3 <- 1 4 
    bat.append(b:bat[:int], 10:int);    	#[3] (0) BKCappend_val_wrap 5 <- 1 6 
    bat.append(b:bat[:int], 20:int);    	#[4] (0) BKCappend_val_wrap 7 <- 1 8 
    r:bat[:int] := mal.multiplex("user":str, "foo":str, b:bat[:int]);	#[5] (0) MANIFOLDremapMultiplex 9 <- 10 11 1 
    io.print(r:bat[:int]);              	#[6] (0) IOprint_val 12 <- 9 
end user.qry;                           	#[7] (0)  

# 11:53:01 >  
# 11:53:01 >  Done.
# 11:53:01 >  

